### PR TITLE
Restore 'id' stanza in pull results

### LINKED ISSUE #7686: APIv2 pull: response body is different from the DOC
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**
/kind bug

**Description**
I just pull the latest code and build podman. When I use pull feature by APIv2, I got a different response body than the DOC says - http://docs.podman.io/en/latest/_static/api.html#operation/libpodImagesPull. I tried it this months ago, the result was following the DOC.

**Steps to reproduce the issue:**

1. start APIv2 service

2. curl -X POST http://localhost:$PORT/v1.40/libpod/images/pull\?reference\=quay.io/libpod/alpine_labels:latest

**Describe the results you received:**
```
# curl -X POST http://localhost:8082/v1.40/libpod/images/pull\?reference\=quay.io/libpod/alpine_labels:latest
{"stream":"Trying to pull quay.io/libpod/alpine_labels:latest...\n"}
{"stream":"Getting image source signatures\n"}
{"stream":"Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\n"}
{"stream":"Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\n"}
{"stream":"Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\n"}
{"stream":"Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\n"}
{"stream":"Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\n"}
{"stream":"Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\n"}
{"stream":"Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\n"}
{"stream":"Copying blob sha256:4fe2ade4980c2dda4fc95858ebb981489baec8c1e4bd282ab1c3560be8ff9bde\n"}
{"stream":"Copying blob sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4\n"}
{"stream":"Writing manifest to image destination\n"}
{"stream":"Storing signatures\n"}
{"images":["4fab981df737963d83b548a3a6f4e7b43f718eb88567ba5ec7921f636a93d7bc"]}
```

**Describe the results you expected:**
Months ago the result was:
{"id": "4fab981df737963d83b548a3a6f4e7b43f718eb88567ba5ec7921f636a93d7bc"}

### PR DESCRIPTION
**Pull request by jwhonce**
id is the last image id from the set of id's returned via the images
stanza.

id may be deprecated in a future version of the API

Created test_rest_v2_0_0.py to reflect the bump in the API Version.

Fixes #7686

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: pkg/api/handlers/libpod/images_pull.go [hunk 1]
```diff
@@ -182,6 +182,13 @@ loop: // break out of for/select infinite loop
 				if err := enc.Encode(report); err != nil {
 					logrus.Warnf("Failed to json encode error %q", err.Error())
 				}
+
+				report.Images = nil
+				report.ID = images[len(images)-1]
```
Comment (vrothberg) [inline pkg/api/handlers/libpod/images_pull.go:187]: I think we should only set it if `all == false` (or document which image it refers to).

**Review by vrothberg — COMMENTED** [review]
Other than the nit, LGTM

**Comment by mheon** [comment]
Restarted the failed job. It looks like it was actually working properly and getting close to finishing - it just ran out of time at the end of our 120m jobs. Which is concerning.

**Comment by mheon** [comment]
LGTM once comment from @vrothberg is addressed

**Comment by TomSweeneyRedHat** [comment]
LGTM once @vrothberg 's comment is addressed.

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM

**Pull request merged**